### PR TITLE
NIFI-4819 Added support to delete blob from Azure Storage container

### PR DESCRIPTION
**Pull request by zenfenan**
Implemented basic version that takes care of deleting an Azure blob from Storage container, if the blob exists. Also implemented Test.

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/DeleteAzureBlobStorage.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+        if(flowFile == null) {
+            return;
+        }
+
+        final long startNanos = System.nanoTime();
+        String containerName = context.getProperty(AzureStorageUtils.CONTAINER).evaluateAttributeExpressions(flowFile).getValue();
+        String blobPath = context.getProperty(BLOB).evaluateAttributeExpressions(flowFile).getValue();
+
+        AtomicReference<Exception> storedException = new AtomicReference<>();
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/DeleteAzureBlobStorage.java]: Is it really needed?

File: nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/DeleteAzureBlobStorage.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import java.net.URISyntaxException;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+
+@Tags({ "azure", "microsoft", "cloud", "storage", "blob" })
+@SeeAlso({ ListAzureBlobStorage.class, FetchAzureBlobStorage.class, PutAzureBlobStorage.class})
+@CapabilityDescription("Deletes the provided blob from Azure Storage")
+@InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/DeleteAzureBlobStorage.java]: Just
````java
@InputRequirement(Requirement.INPUT_REQUIRED)
````
to be consistent.

**Comment by zenfenan** [comment]
@pvillard31 Actually I left `AtomicReference` on purpose. The reason was I wanted to implement some features later that would need that but yeah it doesn't concern or matter for the current commit/Jira so I took that off as well as fixed to `@InputRequirement(Requirement.INPUT_REQUIRED)`

**Comment by pvillard31** [comment]
+1, thanks for the changes @zenfenan, merging to master.

**Pull request closed without merging**